### PR TITLE
GEODE-8251: make sure Configuration can be deserialized post 1.12.

### PR DESCRIPTION
**Pull request by jinmeiliao**
Co-authoried-by: Alberto Gomez <alberto.gomez@est.tech>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by jinmeiliao** [comment]
@albertogpz

File: geode-core/src/upgradeTest/java/org/apache/geode/internal/cache/rollingupgrade/RollingUpgradeWithGfshDUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+@RunWith(Parameterized.class)
+@Parameterized.UseParametersRunnerFactory(CategoryWithParameterizedRunnerFactory.class)
+public class RollingUpgradeWithGfshDUnitTest {
+  private final UniquePortSupplier portSupplier = new UniquePortSupplier();
+  private final String oldVersion;
+
+  @Parameterized.Parameters(name = "{0}")
+  public static Collection<String> data() {
+    List<String> result = VersionManager.getInstance().getVersionsWithoutCurrent();
+    result.removeIf(s -> TestVersion.compare(s, "1.10.0") < 0);
```
Comment (albertogpz) [inline geode-core/src/upgradeTest/java/org/apache/geode/internal/cache/rollingupgrade/RollingUpgradeWithGfshDUnitTest.java:70]: Why start from this version? Is the upgrade from previous versions not supported?

Comment (jinmeiliao) [reply geode-core/src/upgradeTest/java/org/apache/geode/internal/cache/rollingupgrade/RollingUpgradeWithGfshDUnitTest.java:70]: There is no change in the the Configuration object except in 1.12, so only to test a few version before it and after it. Start from 1.10 just to save a few test cycles, but we can test the previous ones.

**Comment by jinmeiliao** [comment]
> What's the point for these changes?
> I have ran the test case without them and it passes.
> Also, I have searched in the Geode code for the use of this file and have only found it in `ObjectInputStreamFilterWrapper.java`. From what I see there, only the first part of every line, the class name, is used.
> What am I missing?

the previous version of Deployment has variable name "jarFileName" instead of "fileName", when we are deserializing a previous version bytes, if the variable name doesn't match, it won't take the value. We should have a test to verify that, but I saw the behavior in the debugger.

**Comment by jinmeiliao** [comment]
@albertogpz the pipeline tests failed because we added GEODE_HOME environment variable in all tests environment, some tests are able to find the war files thus starting the web services, thus getting the address bind exception. Is there any way we can add that environment variable only to one test?

**Comment by albertogpz** [comment]
> > What's the point for these changes?
> > I have ran the test case without them and it passes.
> > Also, I have searched in the Geode code for the use of this file and have only found it in `ObjectInputStreamFilterWrapper.java`. From what I see there, only the first part of every line, the class name, is used.
> > What am I missing?
> 
> the previous version of Deployment has variable name "jarFileName" instead of "fileName", when we are deserializing a previous version bytes, if the variable name doesn't match, it won't take the value. We should have a test to verify that, but I saw the behavior in the debugger.

I understand that. What I do not get are the changes in the 

> > What's the point for these changes?
> > I have ran the test case without them and it passes.
> > Also, I have searched in the Geode code for the use of this file and have only found it in `ObjectInputStreamFilterWrapper.java`. From what I see there, only the first part of every line, the class name, is used.
> > What am I missing?
> 
> the previous version of Deployment has variable name "jarFileName" instead of "fileName", when we are deserializing a previous version bytes, if the variable name doesn't match, it won't take the value. We should have a test to verify that, but I saw the behavior in the debugger.

I understand the change in the members of the class. What I do not get is why it needs to be reflected in sanctioned-geode-management-serializables.txt.
Who is accessing that file?

**Comment by jinmeiliao** [comment]
> > > What's the point for these changes?
> > > I have ran the test case without them and it passes.
> > > Also, I have searched in the Geode code for the use of this file and have only found it in `ObjectInputStreamFilterWrapper.java`. From what I see there, only the first part of every line, the class name, is used.
> > > What am I missing?
> > 
> > 
> > the previous version of Deployment has variable name "jarFileName" instead of "fileName", when we are deserializing a previous version bytes, if the variable name doesn't match, it won't take the value. We should have a test to verify that, but I saw the behavior in the debugger.
> 
> I understand that. What I do not get are the changes in the
> 
> > > What's the point for these changes?
> > > I have ran the test case without them and it passes.
> > > Also, I have searched in the Geode code for the use of this file and have only found it in `ObjectInputStreamFilterWrapper.java`. From what I see there, only the first part of every line, the class name, is used.
> > > What am I missing?
> > 
> > 
> > the previous version of Deployment has variable name "jarFileName" instead of "fileName", when we are deserializing a previous version bytes, if the variable name doesn't match, it won't take the value. We should have a test to verify that, but I saw the behavior in the debugger.
> 
> I understand the change in the members of the class. What I do not get is why it needs to be reflected in sanctioned-geode-management-serializables.txt.
> Who is accessing that file?

the `AnalyzeManagementSerializablesJUnitTest` is making sure that we don't make random changes in those serialized classes. Every change in the sanction list poses a potential upgrade issue. So the fact that the test is broken will make the developer think twice about the changes he/she is making to the class.

**Comment by albertogpz** [comment]
> > > > What's the point for these changes?
> > > > I have ran the test case without them and it passes.
> > > > Also, I have searched in the Geode code for the use of this file and have only found it in `ObjectInputStreamFilterWrapper.java`. From what I see there, only the first part of every line, the class name, is used.
> > > > What am I missing?
> > > 
> > > 
> > > the previous version of Deployment has variable name "jarFileName" instead of "fileName", when we are deserializing a previous version bytes, if the variable name doesn't match, it won't take the value. We should have a test to verify that, but I saw the behavior in the debugger.
> > 
> > 
> > I understand that. What I do not get are the changes in the
> > > > What's the point for these changes?
> > > > I have ran the test case without them and it passes.
> > > > Also, I have searched in the Geode code for the use of this file and have only found it in `ObjectInputStreamFilterWrapper.java`. From what I see there, only the first part of every line, the class name, is used.
> > > > What am I missing?
> > > 
> > > 
> > > the previous version of Deployment has variable name "jarFileName" instead of "fileName", when we are deserializing a previous version bytes, if the variable name doesn't match, it won't take the value. We should have a test to verify that, but I saw the behavior in the debugger.
> > 
> > 
> > I understand the change in the members of the class. What I do not get is why it needs to be reflected in sanctioned-geode-management-serializables.txt.
> > Who is accessing that file?
> 
> the `AnalyzeManagementSerializablesJUnitTest` is making sure that we don't make random changes in those serialized classes. Every change in the sanction list poses a potential upgrade issue. So the fact that the test is broken will make the developer think twice about the changes he/she is making to the class.

I see now. Thanks for the clarification.

File: geode-core/src/upgradeTest/java/org/apache/geode/internal/cache/rollingupgrade/RollingUpgradeWithGfshDUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.cache.rollingupgrade;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
```
Comment (kirklund) [inline geode-core/src/upgradeTest/java/org/apache/geode/internal/cache/rollingupgrade/RollingUpgradeWithGfshDUnitTest.java:18]: You accidentally have the license header in this class twice.

File: geode-core/src/upgradeTest/java/org/apache/geode/internal/cache/rollingupgrade/RollingUpgradeWithGfshDUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    ClassBuilder classBuilder = new ClassBuilder();
+    File jarsDir = tempFolder.newFolder();
+    String jarName1 = "DeployCommandsDUnit1.jar";
+    File jar1 = new File(jarsDir, jarName1);
+    String class1 = "DeployCommandsDUnitA";
+    classBuilder.writeJarFromName(class1, jar1);
+    return "deploy --dir=" + jarsDir.getAbsolutePath();
+  }
+
+  protected String startServerCommand(String name, int port, int connectedLocatorPort) {
```
Comment (kirklund) [inline geode-core/src/upgradeTest/java/org/apache/geode/internal/cache/rollingupgrade/RollingUpgradeWithGfshDUnitTest.java:142]: Is there a subclass? I'd like to see these `protected` methods be `private` if possible.

File: geode-management/src/main/java/org/apache/geode/management/configuration/AbstractConfiguration.java [hunk 4]
```diff
@@ -47,7 +47,7 @@
 @Experimental
 public abstract class AbstractConfiguration<R extends RuntimeInfo>
     implements Identifiable<String>, JsonSerializable {
-
+  private static final long serialVersionUID = -6612840641128145954L;
```
Comment (kirklund) [inline geode-management/src/main/java/org/apache/geode/management/configuration/AbstractConfiguration.java:50]: Optional: all JDK classes put serialVersionUID at the bottom of the class. I usually do this to try and be consistent but obviously doesn't matter much.

File: geode-core/src/upgradeTest/java/org/apache/geode/internal/cache/rollingupgrade/RollingUpgradeWithGfshDUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    verifyListDeployed(locatorPort);
+
+    oldGfsh.stopLocator(startupExecution, "loc2");
+    GfshScript.of(startLocatorCommand("loc2", locator2Port, locator2JmxPort, locatorPort))
+        .execute(currentGfsh);
+    verifyListDeployed(locator2Port);
+
+    // make sure servers can do rolling upgrade too
+    oldGfsh.stopServer(startupExecution, "server1");
+    GfshScript.of(startServerCommand("server1", server1Port, locatorPort)).execute(currentGfsh);
```
Comment (jchen21) [inline geode-core/src/upgradeTest/java/org/apache/geode/internal/cache/rollingupgrade/RollingUpgradeWithGfshDUnitTest.java:103]: Better `verifyListDeployed` after rolling upgrade the servers.

File: geode-core/src/upgradeTest/java/org/apache/geode/internal/cache/rollingupgrade/RollingUpgrade2DUnitTestBase.java [hunk 6]
```diff
@@ -1185,6 +1185,8 @@ Properties getLocatorProperties(String locatorsString, boolean enableCC) {
     props.setProperty(DistributionConfig.LOCATORS_NAME, locatorsString);
     props.setProperty(DistributionConfig.LOG_LEVEL_NAME, DUnitLauncher.logLevel);
     props.setProperty(DistributionConfig.ENABLE_CLUSTER_CONFIGURATION_NAME, enableCC + "");
+    // do not start http service to avoid port conflict between upgrade tests
+    props.setProperty(DistributionConfig.HTTP_SERVICE_PORT_NAME, "0");
```
[the remark this answers is not in the document]
Comment (jinmeiliao) [reply geode-core/src/upgradeTest/java/org/apache/geode/internal/cache/rollingupgrade/RollingUpgrade2DUnitTestBase.java:1189]: here 0 means do not start.

**Pull request merged**